### DISCUSSION TITLE
1.8.6

### DISCUSSION THREAD

**Discussion by vvb2060** [LSPosed/LSPosed#2337] [New Release]
### New Xposed API proposal
As Android version iterates, the original Xposed API by rovo89 reaches its limits. Now we are working on the new modern Xposed API with features of application scope management, remote preferences, dex parser interface and so on.
The new API will be implemented in the next releasing of LSPosed, and it is welcome to post your suggestions on https://github.com/libxposed.

### New way to open parasitic manager
Some devices cannot create shortcuts due to kernel issues, now LSPosed switches to opening parasitic manager from notification. This notification is always present and can be disable in manager settings.

### Changelog
- Some manager UI fixes
- Update DoH
- Set the dex2oat wrapper owner and group to root:shell
- Guard backup during hook
- Add notification to open parasitic manager
- Fix hook 32bit process
- Fix dex2oat fallback
- Fix webview permission

### 新的 Xposed API 提案
随着 Android 版本的迭代，原有的 rovo89 Xposed API 已经达到了极限。 我们正在开发新的现代 Xposed API，它具有模块作用域管理、远程配置文件、dex 解析器接口等功能。
新的 API 将在 LSPosed 的下一个版本中实现，欢迎在 https://github.com/libxposed 上发表您的建议。

### 新的管理器打开方式
有部分设备因内核问题无法创建快捷方式，现在 LSPosed 切换到从通知打开管理器。此通知一直存在，也可以在管理器设置中关闭通知。

### 更新日志
- 一些管理器界面修复
- 更新安全 DNS 实现
- 将 dex2oat 包装器权限设置为 root:shell
- hook 期间保护备份
- 添加从通知打开寄生管理器的方式
- 修复 hook 32 位进程
- 修复 dex2oat 回退
- 修复 webview 文件权限

### Contributors
@yujincheng08 @vvb2060 @Howard20181 @Dr-TSNG

<hr /><em>This discussion was created from the release <a href='https://github.com/LSPosed/LSPosed/releases/tag/v1.8.6'>1.8.6</a>.</em>

**Comment by MentalEr15** [comment]
remove the permanent lspozed job notification! although it can be hidden, but after a reboot, it returns! infuriating! even without a reboot, it returns from time to time, despite disabling the notification of your thing. this is exactly in the 1.8.6 version of such stuff.

**Reply by juancavargas5** [reply]
Hl tengo un grave problema es que no sé cómo activar lsposed y abofahdsh

**Reply by juancavargas5** [reply]
Me podrían ayudar

**Reply by juancavargas5** [reply]
En general tengo un teléfono de gana baja redmi 9c

**Comment by Howard20181** [comment]
Please read changelog!
It can be disable in LSPosed manager.


Best regards

,
Howard
________________________________

**Reply by yonifrigiawan** [reply]
after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Comment by Redeorhcs** [comment]
Do any special steps need to be taken to install version 1.8.6 if we are currently on the stable version? 1.8.5? Do modules need to be disabled? Does anything need to be toggled on or off in magisk? Do we just download from within the LS posed app , install, and reboot?

**Reply by MentalEr15** [reply]
just update via the app. it will do everything by itself.

**Comment by Redeorhcs** [comment]
Thanks

**Comment by Redeorhcs** [comment]
Should the module be updated through the majisk manager application or the LS posed application?

**Comment by uragiristereo** [comment]
What a great release, thanks for still providing Riru module version! My banking app can detect root if i use Zygisk but it's not the case with Riru.

**Comment by TheHippoMan** [comment]
After upgrading to the Stable riru version of LSPosed 1.8.6 on my Pixel 5 running Android 11 and Magisk 23.0, I am not getting any persistent LSPosed notification, even when I go into the LSPosed settings and enable that notification by turning on the "Status Notification" setting.

I uninstalled LSPosed 1.8.6 and re-installed it, and I still am not seeing that Notification.

LSPosed is running with no problem. The only issue is the missing Status Notification.

Is there something I need to do beyond the normal installation instructions in order to enable this Status Notification?

Thank you very much in advance.

**PS**: To be clear, I do **_not_** have the shortcut created. Given that there is no notification, the only way I can invoke the parasitic manager for version 1.8.6 is via the `*#*#5776773#*#*` USSD sequence.

**PPS**:  And even when I **_do_** have the shortcut created and turn the "Status Notification" setting **ON**, I am still not getting any status notification.

**Comment by BROKEN81** [comment]
Is this normal for LSPosed? Hold for me, spam and call screen, direct my call getting disabled ( options completely gone). 
I'm using a Pixel 6 Pro on latest 3.1 build.

**Comment by LF1412** [comment]
666

**Comment by Tempzxxx** [comment]
更新riru版1.8.6后，状态栏一直有通知显示，且无法移除。望改进。

**Reply by TheHippoMan** [reply]
> 更新riru版1.8.6后，状态栏一直有通知显示，且无法移除。望改进。

In my case, it's the opposite situation: with version 1.8.6, there **_never_** (!!) is any LSPosed notification in the status bar. For more details, please see my message above that was written on January 21, 2023. Thank you.

**Comment by laup1933** [comment]
Thanks for the work!  but this module blocks some applications like camera, flashlight and others.  I use an infinix hot 12

**Comment by Taevoloev** [comment]
Android 14 beta 1 cannot open the manager, pixel 6

**Reply by amritpal2489** [reply]
> Android 14 beta 1 cannot open the manager, pixel 6

Install the manager from /data/adb/lspd

**Reply by amritpal2489** [reply]
> Android 14 beta 1 cannot open the manager, pixel 6

Did you find any other solution?

**Reply by Taevoloev** [reply]
install 1.8.6 （6884）![Screenshot_20230423-121536.png](https://user-images.githubusercontent.com/124876450/233819616-6c597c5c-7e22-4bd1-9fa0-250913fe2290.png)

**Reply by yonifrigiawan** [reply]
after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Reply by yonifrigiawan** [reply]
> > Android 14 beta 1 cannot open the manager, pixel 6
> 
> Install the manager from /data/adb/lspd

after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Comment by H811566880** [comment]
总是管理器奔溃
Build version: 1.8.6 (6712) 
Current date: 2023-04-15T21:47:04.148517 
Device: Xiaomi M2102J2SC 
Fingerprint: Xiaomi/thyme/thyme:13/TKQ1.220829.002/V14.0.4.0.TGACNXM:user/release-keys 
 
SDK: 33 
 
Stack trace:  
java.lang.RuntimeException: Error receiving broadcast Intent { act=org.lsposed.manager.NOTIFICATION flg=0x1400010 pkg=org.lsposed.manager (has extras) } in a2@94a07d1
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1832)
	at android.app.LoadedApk$ReceiverDispatcher$Args$$ExternalSyntheticLambda0.run(Unknown Source:2)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:210)
	at android.os.Looper.loop(Looper.java:299)
	at android.app.ActivityThread.main(ActivityThread.java:8261)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.pm.PackageInfo Sh.h(int, java.lang.String)' on a null object reference
	at am.f(Unknown Source:29)
	at a2.onReceive(Unknown Source:151)
	at android.app.LoadedApk$ReceiverDispatcher$Args.lambda$getRunnable$0$android-app-LoadedApk$ReceiverDispatcher$Args(LoadedApk.java:1819)
	... 9 more

**Comment by Taevoloev** [comment]
No

**Comment by yonifrigiawan** [comment]
after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Comment by yonifrigiawan** [comment]
after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Comment by yonifrigiawan** [comment]
after I installed lposed via magisk delta , the lposed app didn't appear. how to solve it?

**Comment by hexman1999** [comment]
Implementing some modules to specific apps makes xposed hooks visible 

I am trying to hide developer options from (faisal cash) app using DevOptsHide module which actually hides it but then the app detects xposed hooks and doesn't open

**Comment by Barkuti** [comment]
I have to agree with **MentalEr15**, author of the first reply to this thread:

![Screenshot_System_UI_20230602-070059](https://github.com/LSPosed/LSPosed/assets/28562086/b8972f44-efcf-47b9-843c-199f846ef5ad)
![Screenshot_System_UI_20230602-070104](https://github.com/LSPosed/LSPosed/assets/28562086/20918a13-b0db-43f0-bd24-4c6d552ceb44)

**What the hell?** You set the LSPosed notification like a pesky “Android System” notification which I cannot KILL? FOR FUCKS' SAKE!
There's nothing which I hate more in an Android OS or device than a pesky notification which I cannot remove from the pull-down menu, nothing!
Yes, already read that it can be disabled in the manager. **But why?** Why does it has to be this way?

![Screenshot_Settings_20230602-072846](https://github.com/LSPosed/LSPosed/assets/28562086/a5feda14-4c25-47c8-bfaa-35b64dfba004)

That is part of the **pretty long** list of notifications assigned to “Android System” in my device OS (AEX v5.7, Android 8.1), and a good chunk of them cannot be disabled and I don't @#$% care whether they are considered “important” or not, I JUST WANT TO @#$% THEM. Yes, I know this is somewhat off-topic from this, but this has put the finger on the sore spot… 😒😒😒

Why does LSposed has to be this way? This is nonsense, in my experience only certain applications complain about custom ROMs or rooted systems, but nothing about LSposed. After all, to run LSPosed you must first have a rooted Android OS anyway, for fucks' sake.
Why am I saying this? Heck, make LSPosed install a shortcut in the launcher. **Problem solved.** And if someone wants to remove LSPosed from the launcher, just let them; I just don't see any reason for such a thing.
If LSPosed is meant to be any sort of furtive application, setting it to launch via a permanent Android notification in the drop down menu is… like MentalEr15 said **infuriating**, and quite the opposite.

Regards.

P.S.: went back to version 1.8.4.

**Comment by ixyzi2005** [comment]
闪退
